src/util/compileProject.js

```diff
@@ -156,16 +154,28 @@ async function addPreviewSupportScript(doc) {
   doc.head.appendChild(scriptTag);
 }
 
-async function addJavascript(
+export async function addJavascript(
   doc,
   {sources: {javascript}},
-  {breakLoops = false},
+  opts,
 ) {
   if (trim(javascript).length === 0) {
     return;
   }
 
-  let source = `\n${sourceDelimiter}\n${javascript}`;
+  const {breakLoops} = opts || {};
+
+  let source = javascript;
+
+  const {babelWithEnv} = await retryingFailedImports(
+    () => import(
+      /* webpackChunkName: "babel" */
+      '../services/babel-browser.gen' // eslint-disable-line comma-dangle
+    ),
+  );
+
+  source = await babelWithEnv(source);
```

Comment:
Seems like we could just do `const source = await babelWithEnv(javascript)` here? (No need to reassign `source`)